homeassistant/components/image_processing/facebox.py

```diff
@@ -63,16 +74,55 @@ def parse_faces(api_faces):
     return known_faces
 
 
+def post_image(url, image):
+    """Post an image to the classifier."""
+    try:
+        response = requests.post(
+            url,
+            json={"base64": encode_image(image)},
+            timeout=TIMEOUT
+            )
+        return response
+    except requests.exceptions.ConnectionError:
+        _LOGGER.error("ConnectionError: Is %s running?", CLASSIFIER)
+
+
+def valid_file_path(file_path):
+    """Check that a file_path points to a valid file."""
+    try:
+        cv.isfile(file_path)
+        return True
+    except vol.Invalid:
+        _LOGGER.error(
+            "%s error: Invalid file path: %s", CLASSIFIER, file_path)
+        return False
+
+
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the classifier."""
     entities = []
+    facebox = None
     for camera in config[CONF_SOURCE]:
-        entities.append(FaceClassifyEntity(
+        facebox = FaceClassifyEntity(
             config[CONF_IP_ADDRESS],
             config[CONF_PORT],
             camera[CONF_ENTITY_ID],
-            camera.get(CONF_NAME)
-        ))
+            camera.get(CONF_NAME))
+
+        def teach_service(call):
+            """Teach a facebox a name."""
+            name = call.data.get(ATTR_NAME)
+            file_path = call.data.get(FILE_PATH)
+            facebox.teach(name, file_path)
+            return True
+
+        hass.services.register(
```

Comment:
This won't work. Only one service is registered per domain and service name key. Move this out of the for loop.

Reply:
ok